### DISCUSSION TITLE
.getConnectedDevice() not working

### DISCUSSION THREAD

**Discussion by Iamsabarinarayanan** [kenjdavidson/react-native-bluetooth-classic#234] [Q&A]
Thanks for this amazing library but I was not able to list the connected device using any @kenjdavidson
 
```
try {
    // Getting all paired/bonded devices
    const paired = await RNBluetoothClassic.getBondedDevices();

    // tried connecting a device outside react native app, device is connected
    const connected = await RNBluetoothClassic.getConnectedDevices();
                                                                                                                        
     const connection= await paired[0].connect()

    console.log("connected device:\n",connected);--> empty array
     console.log("connection:\n",connection);---> no log is printed   
} catch (err) {
    // Error if Bluetooth is not enabled
    // Or there are any issues requesting paired devices
}}
```

is there any sample for react native documentation using react hooks and functional component?

Your help will be much appreciated here:)

**Comment by kenjdavidson** [comment]
From the code it looks like you're:
a) doing things out of order
b) not fully understanding what the methods are for

Specifically this comment `// tried connecting a device outside react native app, device is connected` is curious.  I'm not sure what you mean by connecting a device outside the app.

To break it down:

`getBondedDevices` is used to return paired/bonded devices, it looks like you're using this correctly.
`getConnectedDevices` is used to return devices which currently have an active RFCOMM socket open and communication is available with the device.  Personally, I don't ever use this, but it's available incase you (for some reason) don't maintain the list of connected devices after you've initiated a connection.

Essentially you want the order to be:
- get paired devices (store to state or store)
- connect to selected device 
- use connected device

Now if you really need to use `getConnectedDevices` then you can do so after you've connected to one.  There are a number of example apps linked to this project (look for the `projects using` link on the main page, I'm sure one of them has examples of functional components and hooks.

The BluetoothClassicExample app that I have / use for development is pretty simple but may provide you with examples.  There is no difference in using hooks, it's just how you decide to use React.

**Reply by Iamsabarinarayanan** [reply]
Hi @kenjdavidson ,

Thanks for the reply. Now I was able to connect with other devices however I came up with another constrain..

When I tried connecting to my test device Galaxy M21, It provided me below promise rejection. however pairing request is working with the device but successful connection cannot be established.

WARN  Possible Unhandled Promise Rejection (id: 0):
Error: java.io.IOException: read failed, socket might closed or timeout, read ret: -1
Error: java.io.IOException: read failed, socket might closed or timeout, read ret: -1 

It happens only with this device , May I know what could be the possible reason for it?
